### DISCUSSION TITLE
Layers - Invalid javascript condition behavior

### DISCUSSION THREAD

**Discussion by sobotama** [OrchardCMS/OrchardCore#12894] [Q&A] [answered]
Edit: opened issue https://github.com/OrchardCMS/OrchardCore/issues/12901#issue-1467671906

Hi, I have a question about the handling of invalid javacript conditions in layer rules. 
One of our users writes javascript conditions in layer rules for resolving URLs and once they mistyped and wrote the condition like this: 

`("test")`

instead of this:

`url("test")`

After adding any widget to this faulty layer the whole frontend fails with code 500: 

`FormatException: String 'test' was not recognized as a valid Boolean.`
`bool.Parse(ReadOnlySpan<char> value)`
`bool.Parse(string value)`
`string.System.IConvertible.ToBoolean(IFormatProvider provider)`
`System.Convert.ToBoolean(object value)`
`OrchardCore.Rules.Services.JavascriptConditionEvaluator.EvaluateAsync(JavascriptCondition condition) in JavascriptConditionEvaluator.cs`
the rest of stack is ommited.

Backend is still accesible and it can be repaired, however the client is unhappy, that the whole site can be broken this easily, even though they now use the proper URL condition. 

Is this the intended behavior? Or wouldn't it be better for it to fail in the background and log it, without the whole frontend crashing?

**Comment by Skrypt** [comment, marked as answer]
Please open an issue if not already.
